crates/lib/src/executable.rs

```diff
@@ -329,7 +329,7 @@ impl Executable<'_, '_> {
     /// To disable compilation, set this to `None`.
     #[must_use]
     #[allow(trivial_casts)]
-    pub fn with_quilc_client<C: quilc::Client + Send + Sync + 'static>(
+    pub fn with_quilc_client<C: quilc::Client<Error = quilc::Error> + Send + Sync + 'static>(
```

Comment:
Is it possible/preferrable to make this constraint a `quilc::Client<Error = Into<quilc::Error>>`? I get an error trying to use the libquil client, see https://github.com/rigetti/qcs-sdk-rust/pull/387